File: py/day1.py
# day 1
raw = open('./data/1.txt').read().strip()
data = [int(x) for x in raw.split('\n')]

# p1
print(sum([1 for i in range(1, len(data)) if data[i] > data[i-1]]))

# p2
print(sum([1 for i in range(len(data)-3) if data[i+3] > data[i]]))

File: js/day1.js
// day 1
fetch('data/1.txt')
.then(s => s.text())
.then(s => {
  s = s.trim().split('\r\n').map(x => parseInt(x));
  console.log(s.map((v, i) => s[i-1] < v).reduce((a, b) => a + b));
  console.log(s.map((v, i) => s[i+3] > v).reduce((a, b) => a + b));
});
